File: Distinct Numbers/Distinct_Numbers.cpp
#include<bits/stdc++.h>
using namespace std;
 
#define ll long long
#define int ll
 
signed main()
{
    ios::sync_with_stdio(false);
    cin.tie(NULL);
    
    int len, i, ans = 0;
    cin >> len;
    int arr[len];
    map<int, int> mii;
    for (i = 0; i < len; i++)
    {
        cin >> arr[i];
        mii[arr[i]]++;
        if(mii[arr[i]] == 1)
        {
            ans++;
        }
    }
    cout << ans;
}

File: Distinct Numbers/Distinct_Numbers.py
def main():
    length = int(input().strip())
    arr = list(map(int, input().split()))
    mii = {}
    ans = 0
    
    for num in arr:
        if num not in mii:
            mii[num] = 1
            ans += 1
        else:
            mii[num] += 1
    
    print(ans)


if __name__ == "__main__":
    main()
